python/S3_OBJECT_LOCK_ENABLED/S3_OBJECT_LOCK_ENABLED.py

```diff
@@ -0,0 +1,184 @@
+
+Rule Parameters:
+  Mode
+    The Object Lock retention mode you want to apply to new objects placed in the specified bucket
+  Days
+    The number of days that you want to specify for the default retention period.
+  Years
+    The number of years that you want to specify for the default retention period.
+
+Scenarios:
+  Scenario: 1
+    Given: No S3 buckets available for the account
+     Then: Return NOT_APPLICABLE
+  Scenario: 2
+    Given: S3 bucket is not object lock enabled
+     Then: Return NON_COMPLIANT
+  Scenario: 3
+    Given: S3 bucket is object lock enabled with mode
+     Then: Return COMPLIANT
+  Scenario: 4
+    Given: S3 bucket is object lock enabled with mode and days
+          (actual bucket days should be greater than the parameter days)
+     Then: Return COMPLIANT
+  Scenario: 5
+    Given: S3 bucket is object lock enabled with mode and years
+          (actual bucket years should be greater than the parameter years)
+     Then: Return COMPLIANT
+  Scenario: 6
+    Given: S3 bucket is object lock enabled with mode, days and yeaas
+          (actual bucket days(days + (years*365)) should be greater than the parameter days
+     Then: Return COMPLIANT
+  Scenario: 7
+    Given: S3 bucket is object lock enabled with different mode
+     Then: Return NON_COMPLIANT
+  Scenario: 8
+    Given: S3 bucket is object lock enabled with mode and days
+          (actual bucket days are less than the parameter days)
+     Then: Return NON_COMPLIANT
+  Scenario: 9
+    Given: S3 bucket is object lock enabled with mode and years
+          (actual bucket years are less than the parameter years)
+     Then: Return NON_COMPLIANT
+  Scenario: 10
+    Given: S3 bucket is object lock enabled with mode, days and yeaas
+          (actual bucket days(days + (years*365)) are less than the parameter days)
+     Then: Return NON_COMPLIANT
+'''
+
+import json
+
+from rdklib import Evaluator, Evaluation, ConfigRule, ComplianceType, InvalidParametersError
+
+RESOURCE_TYPE = 'AWS::S3::Bucket'
+
+class S3_OBJECT_LOCK_ENABLED(ConfigRule):
+    def evaluate_periodic(self, event, client_factory, valid_rule_parameters):
+        s3_client = client_factory.build_client("s3") # AWS S3 Client
+        config_client = client_factory.build_client('config') # AWS Config Service Client
+        evaluations = []
+        valid_rule_parameters = self.evaluate_parameters(valid_rule_parameters)
+        for s3_bucket_name in get_buckets(config_client):
+
+            isexception = False
+            try:
+                response = s3_client.get_object_lock_configuration(Bucket=s3_bucket_name)
+            except:
+                isexception = True
+            if isexception or 'ObjectLockConfiguration' not in response:
+                evaluations.append(
+                    Evaluation(ComplianceType.NON_COMPLIANT,
+                               s3_bucket_name, RESOURCE_TYPE,
+                               "No ObjectLockConfiguration exist for this bucket.")
```

Comment:
Does it throw any exception ? if yes In what case it throws any exception can we handle the specific scenarios instead just saying NON_COMPLIANT